paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/http/handlers/EventUtils.java

```diff
@@ -93,6 +100,20 @@ private EventUtils() { }
         }
     }
 
+    private static boolean isEventOfValidType(final JsonNode element, final EventType eventType) {
+        if (!element.isNull() && element.has("metadata")) {
+            if (!element.get("metadata").has("event_type")
+                    || element.get("metadata").get("event_type").asText().equals(eventType.getName())) {
+                return true;
+            } else {
+                throw new InvalidEventTypeException(format("Unexpected event type (expected=[%s], actual=[%s])",
+                        eventType.getName(), element.get("metadata").get("event_type").asText()));
+            }
+        }
+
+        return false;
```

Comment:
This method is strange:

1.  Shouldn't it be `true` here? `metadata` and `event_type` are not required fields according to the nakadi API.
1. return value can be `false` and sometimes a `InvalidEventTypeException` is thrown. However, according to the method and exception name, these are the same thing.